src/Compilers/CSharp/Test/Semantic/Semantics/AccessCheckTests.cs

```diff
@@ -1478,5 +1502,26 @@ static void Main()
                 //         new A();
                 Diagnostic(ErrorCode.ERR_BadAccess, "A").WithArguments("A").WithLocation(5, 13));
         }
+
+        [Fact]
+        public void FunctionPointerTypesFromOtherInaccessibleAssembly()
+        {
+            var comp = CreateCompilation(@"
+unsafe class A
+{
+    internal delegate*<A> ptr1;
+    internal delegate*<A, object> ptr2;
+}", options: TestOptions.UnsafeReleaseDll, parseOptions: TestOptions.RegularPreview);
+
+            var ptr1 = comp.GetMember<FieldSymbol>("A.ptr1").Type.GetPublicSymbol();
+            var ptr2 = comp.GetMember<FieldSymbol>("A.ptr2").Type.GetPublicSymbol();
+
+            var comp2 = CreateCompilation("class B {}");
+
+            var b = comp2.GetMember("B").GetPublicSymbol();
+
+            Assert.Throws<ArgumentException>(() => ((Compilation)comp2).IsSymbolAccessibleWithin(ptr1, b));
```

Comment:
>IsSymbolAccessibleWithin [](start = 72, length = 24)

Why does this throw `ArgumentException` rather than returning `false`?

Reply:
Because `A` isn't from any reference of `comp2`.

---
In reply to: [428179712](https://github.com/dotnet/roslyn/pull/44257#discussion_r428179712) [](ancestors = 428179712)